src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -942,19 +942,16 @@ private static BoundExpression MakeLiteral(SyntaxNode syntax, ConstantValue cons
 
         private static void OptimizeTemporaries(
             BoundExpression[] arguments,
-            ArrayBuilder<RefKind> refKinds,
             ArrayBuilder<BoundAssignmentOperator> storesToTemps,
             ArrayBuilder<LocalSymbol> temporariesBuilder)
         {
             Debug.Assert(arguments != null);
-            Debug.Assert(refKinds != null);
```

Comment:
Should we keep those assertions, just moving them to the caller (`MakeArguments`)?

Reply:
I am not sure the assertion was very useful. It just asserts that the builder exists (which would cause an obvious NRE anyways).
The new asserts in emit should catch inconsistencies in the actual content of ref kinds, so I think we do not need to keep this one.

---
In reply to: [160823005](https://github.com/dotnet/roslyn/pull/24128#discussion_r160823005) [](ancestors = 160823005)